src/main/java/seedu/address/model/transaction/Description.java

```diff
@@ -0,0 +1,41 @@
+package seedu.address.model.transaction;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Transaction's Description in the Wallet.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDescription(String)}
+ */
+public class Description {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Descriptions should not be blank";
+
+    public final String description;
+
+    /**
+     * Constructs a {@code Description}.
+     *
+     * @param description A valid description
+     */
+    public Description(String description) {
+        requireNonNull(description);
+        checkArgument(isValidDescription(description), MESSAGE_CONSTRAINTS);
+        this.description = description;
+    }
+
+    /**
+     * Returns true if a given string is a valid description.
+     */
+    public static boolean isValidDescription(String test) {
+        return test != null && !test.trim().isEmpty();
+    }
+
```

Comment:
Just 1 line of space between methods will do.

Reply:
Oh ya good catch!